packages/components/bucket-policy/create-bucket-policy.tsx

```diff
@@ -0,0 +1,179 @@
+import * as React from 'react';
+import {
+  getAPIVersionForModel,
+  k8sCreate,
+} from '@openshift-console/dynamic-plugin-sdk';
+import * as _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { RouteComponentProps } from 'react-router';
+import { Title } from '@patternfly/react-core';
+import {
+  CACHE_ANN,
+  OBC_NS_ANN,
+  DataResourceType,
+  DATA_FEDERATION_NAMESPACE,
+  EDIT_DATA_RESOURCES,
+} from '../../constants';
+import { NooBaaBucketClassModel } from '../../models';
+import { referenceForModel } from '../../utils';
+import { useModalLauncher } from '../../utils/modals/modalLauncher';
+import { getName } from '../../utils/selectors/k8s';
+import { BucketPolicyBody } from './bucket-policy-body';
+import { BucketPolicyFooter } from './bucket-policy-footer';
+import {
+  bucketPolicyReducer,
+  bucketPolicyInitialState,
+  BucketPolicyState,
+  BucketPolicyActionType,
+} from './state';
+import './bucket-policy.scss';
+
+const extraMap = {
+  [EDIT_DATA_RESOURCES]: React.lazy(
+    () => import('../modals/data-resource/namespace-store-modal')
+  ),
+};
+
+const getPayload = (state: BucketPolicyState, ns: string) => {
+  const metadata = {
+    apiVersion: getAPIVersionForModel(NooBaaBucketClassModel),
+    kind: NooBaaBucketClassModel.kind,
+    metadata: {
+      name: state.bucketName,
+      namespace: ns,
+      annotations: {
+        [CACHE_ANN]:
+          state.dataResourceType === DataResourceType.SINGLE &&
+          state.cacheEnabled
+            ? 'true'
+            : 'false',
```

Comment:
```suggestion
          state.dataResourceType === DataResourceType.SINGLE &&
          state.cacheEnabled,
```

Reply:
annotation takes only string that is why I had to do it explicitly...